### PR TITLE
Introduce molar mass for better balance of SI units

### LINKED ISSUE #4080: Unit error in Modelica.Mechanics.MultiBody.Examples.Loops.Utilities.GasForce2
For `R_air`:
```
  constant SI.SpecificHeatCapacity R_air = Modelica.Constants.R / 0.0289651159;
```
we should introduce the molar mass of air, if I am not mistaken.

### PR DESCRIPTION
**Pull request by tobolar**
Close #4080

### CODE REVIEW

**Comment by TManikantan** [comment]
@MartinOtter @casella @HansOlsson kindly look into the fix for this ticket.

File: Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo [hunk 1]
```diff
@@ -24,7 +24,8 @@ model GasForce2 "Rough approximation of gas force in a combustion engine's cylin
   SI.Velocity v_rel "Relative piston velocity (<0: compression;  >0: expansion)";
 
 protected
-  constant SI.SpecificHeatCapacity R_air = Modelica.Constants.R/0.0289651159;
+  constant SI.MolarMass MM_gas = 0.0289651159;
```
Comment (HansOlsson) [inline Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo:27]:
```suggestion
  constant SI.MolarMass MM_gas = 0.0289651159 "Molar mass for dry air; should be changed";
```
This is somewhat complicated. Just having the value for "air" and calling it "gas" without any explanation seems confusing to me.
I see the following possibilities:

- Explain it as proposed.
- Call it "air" as before - but then we get the question of why there's just air in a combustion chamber.
- Call it "gas" without explanation - but using the name "gas" for "air" seems as if we know it is bad, and just try to hide it.

Comment (tobolar) [reply Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo:27]: @HansOlsson I fully agree with you. To rename to "gas" was just to be complementary with "gasForce". 
To explain the model as purposed, we need the proper molar mass of the gas. This need some recherche. And, of course, the simulation results will change then.
But I agree this is the way we should preferre.

**Comment by TManikantan** [comment]
@tobolar were you able to come up with something considering the suggestions of @HansOlsson.Kindly commit the fix if you have worked it out. Many Thanks

File: Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo (hunk 1 above)
Comment (tobolar) [reply Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo:27]: I reviewed the model and, actually, `R_air / R_gas` is used just to calculate the gas' density `dens` which itself is not further used. So I tend to remove the variables `R_air` and `dens` completely. The regression results shall not be concerned.

**Comment by TManikantan** [comment]
@tobolar were you able to work through the problem?Kindly commit if you have found the fix.Many Thanks..

**Comment by HansOlsson** [comment]
> > were you able to work through the problem?
> 
> See my last comment [#4081 (comment)](https://github.com/modelica/ModelicaStandardLibrary/pull/4081#discussion_r1182312487) I would appreciate if @HansOlsson and @MartinOtter could comment as well.

I agree with that idea - but I guess the thumb-up was a bit too subtle.

File: Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo (hunk 1 above)
Comment (MartinOtter) [reply Modelica/Mechanics/MultiBody/Examples/Loops/Utilities/GasForce2.mo:27]: > I reviewed the model and, actually, `R_air / R_gas` is used just to calculate the gas' density `dens` which itself is not further used. So I tend to remove the variables `R_air` and `dens` completely. The regression results shall not be concerned.

Yes, this makes sense: Remove R_air and dens.

**Comment by tobolar** [comment]
>  the thumb-up was a bit too subtle

@HansOlsson Sorry, I really overlooked it. :-O

**Comment by tobolar** [comment]
Both `dens` and `R_air`deleted now. And removed `dens` from comparison signals of ModelicaTest.

**Comment by TManikantan** [comment]
@MartinOtter @HansOlsson would you please approve the PR as the comments are addressed?

**Pull request merged**